app/src/main/java/ch/epfl/polychef/pages/EntryPage.java

```diff
@@ -56,15 +63,62 @@ protected void onResume() {
     }
 
     protected void goHomeIfConnected() {
-        if(FirebaseAuth.getInstance().getCurrentUser() != null) {
-            startActivity(new Intent(this, HomePage.class));
+        if(getFireBaseAuth().getCurrentUser() != null) {
+            findViewById(R.id.autoLoginBackground).setVisibility(View.VISIBLE);
+            findViewById(R.id.autoLoginProgress).setVisibility(View.VISIBLE);
+
+            preventInteractions();
+
+            Toast.makeText(this, "Auto connect" , Toast.LENGTH_SHORT).show();
+
+            getUserStorage().initializeUserFromAuthenticatedUser(this);
         }
     }
 
+    private void preventInteractions(){
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
+                WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+    }
+
+    private void allowInteractions(){
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+    }
+
     /** Called when the user taps the log button. */
     public void login(View view) {
         Intent intent = new Intent(this, LoginPage.class);
         startActivity(intent);
     }
 
+    @Override
+    public void onFailure() {
+
```

Comment:
Is it ok to just not to handle failures ?

Reply:
Good point, I think it would have kept loading indefinitely. I'll change that